CONTRIBUTING.md

```diff
@@ -3,21 +3,86 @@ Thanks for your interest in contributing to the project! Please follow these sim
-- Please don't add #Regions to the code
-- When adding new files, please prepend the standard license disclaimer (just copy and paste from another source file)
-- Do your best to follow these guidelines but don't be afraid to make mistakes when trying to apply them. We are all novices in the beginning.
+### `CefSharp` vs `CEF`
+
+`CefSharp` simply takes the building blocks provided by `Cef` and attempts to provide a usable '.Net' implementation.
+The upstream [`CEF` forum](http://magpcss.org/ceforum/) is a valuable resource, if your issues seems fairly low level, then please conduct some research before posting.
+
+It maybe helpful to run the `cefclient` application and compare output with `CefSharp`. The `WinForms` and `WPF` versions use two different rending modes, `WPF` uses Offscreen Rendering (`OSR`).
+
+- Download **Test App** from http://cefbuilds.com
+- To compare with WPF run `cefclient --multi-threaded-message-loop --off-screen-rendering-enabled`
+- To compare with WinForms `cefclient --multi-threaded-message-loop`
+
+To determine which version of `CEF` your build is running, open `chrome://version` and you'll see a number similar to `3.2062.1898`, in this case `2062` represents the branch which corresponds to http://cefbuilds.com/#branch_2062
+
+### What should I include when creating an `Issue`?
+
+A bug is a _demonstrable problem_ that is caused by the code in the repository. Ideally each issue is a useful resource for references purposes (Don't take offence if someone edits your description).
+
+1. Good bug reports are extremely helpful. The more information you provide, the more likely your issue will be resolved.
+2. Good bug reports shouldn't leave others needing to chase you up for more information. Be sure to include the
+details of your environment.
+3. 'Ask Don't Tell' : Ask how to achieve something, **don't say it's broken** just because you haven't got it working yet!
+4. **Isolate the problem** &mdash; ideally create a reproducible example.
+5. **Include a screencast if relevant** - Is your issue about a design or front end feature or bug? The most helpful thing in the world is if we can *see* what you're talking about. Use [LICEcap](http://www.cockos.com/licecap/) to quickly and easily record a short screencast (24fps) and save it as an animated gif! Embed it directly into your GitHub issue.
+6. **Include as much info as possible!** Use the **Bug Report template** below or [click this link](https://github.com/CefSharp/CefSharp/issues/new?title=Bug%3A&body=%23%23%23%20Issue%20Summary%0A%0A%23%23%23%20Steps%20to%20Reproduce%0A%0A1.%20This%20is%20the%20first%20step%0A%0AThis%20is%20a%20bug%20because...%0A%0A%23%23%23%20Technical%20details%0A%0A*%20Ghost%20Version%3A%20master%20-%20latest%20commit%3A%20%20INSERT%20COMMIT%20REF%0A*%20Client%20OS%3A%20%0A*%20Server%20OS%3A%20%0A*%20Node%20Version%3A%20%0A*%20Browser%3A%20%0A*%20Database%3A) to start creating a bug report with the template automatically.
```

Comment:
This link seems to be for another project.  Here is a link that encodes the Bug Report template from below:

https://github.com/CefSharp/CefSharp/issues/new?title=Bug%3A&body=**What%20is%20the%20expected%20output%3F%20What%20do%20you%20see%20instead%3F**%0A%0A**What%20steps%20will%20reproduce%20the%20problem%3F**%0A%0A**What%20version%20of%20the%20product%20are%20you%20using%3F%20On%20what%20operating%20system%3F%20x86%20or%20x64%3F**%0A%0A-%20What%20version%20are%20you%20using%3F%20Nuget%3F%20CI%20Nuget%3F%20build%20from%20a%20branch%3F%20If%20so%20which%20branch%3F%0A%0A-%20Win7%2C%20Win%208%2C%20etc%3F%0A%0A**Please%20provide%20any%20additional%20information%20below.**%0A%0A-%20A%20stack%20trace%20if%20available%2C%20any%20Exception%20information.%0A%0A-%20Does%20the%20cef%20log%20provide%20any%20relevant%20information%3F%20(By%20default%20there%20should%20be%20a%20debug.log%20file%20in%20your%20bin%20directory)%0A%0A-%20Any%20other%20background%20information%20that%27s%20relevant%3F%20Are%20you%20doing%20something%20out%20of%20the%20ordinary%3F%203rd%20party%20controls%3F